backend/src/AppServer.hs

```diff
@@ -174,9 +177,33 @@ getLeaderboardHandler year =
 adminRenamePlayerHandler :: RenamePlayerRequest -> AppM NoContent
 adminRenamePlayerHandler RenamePlayerRequest {pid, newName} = runDb $ updatePlayerName pid newName >> pure NoContent
 
+adminModifyReportHandler :: ModifyReportRequest -> AppM NoContent
+adminModifyReportHandler ModifyReportRequest {rid, report} = case validateReport report of
+  Failure errors -> throwError $ err422 {errBody = show errors}
+  Success _ -> runDb $ do
+    oldReport <- readOrError ("Cannot find report " <>: rid) $ lift . get $ rid
+    Entity newWinnerId _ <- readOrError ("Cannot find player " <>: report.winner) $ getPlayerByName report.winner
+    Entity newLoserId _ <- readOrError ("Cannot find player " <>: report.loser) $ getPlayerByName report.loser
+
+    let newReport = toGameReport oldReport.gameReportTimestamp newWinnerId newLoserId report
+    lift $ replace rid newReport
+
+    when (mustReprocess oldReport newReport) reprocessReports
+
+    pure NoContent
+  where
+    mustReprocess :: GameReport -> GameReport -> Bool
+    mustReprocess old new
+      | old.gameReportWinnerId /= new.gameReportWinnerId = True
+      | old.gameReportLoserId /= new.gameReportLoserId = True
+      | old.gameReportSide /= new.gameReportSide = True
+      | old.gameReportCompetition /= new.gameReportCompetition = True
```

Comment:
If `report.competition` changes from [League] -> [League, Tournament], or [Tournament] -> [League], or [League] -> [Tournament], would there be an unnecessary reprocess? `report.match` could be used to avoid that maybs. But also good to err on the side of reprocessing

Reply:
Well.... I definitely *meant* to put match here instead of competition. :face_exhaling: